bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/handler/InsteonDeviceHandler.java

```diff
@@ -533,6 +537,23 @@ public void channelUnlinked(ChannelUID channelUID) {
         logger.debug("channel {} unlinked ", channelUID.getAsString());
     }
 
+    public InsteonAddress getInsteonAddress() {
+        if (config != null) {
+            return new InsteonAddress(config.getAddress());
+        } else {
+            throw new IllegalArgumentException("config is null");
```

Comment:
This is not a correct exception to throw in this case. As documented: "Thrown to indicate that a method has been passed an illegal or inappropriate argument." Instead, you could consider making config `@NonNullByDefault({})` (from `@Nullable`) since it will always be non-null after `initialize()`.

Reply:
changed